scripts/ci/check-dependencies-versions-match.js

```diff
@@ -0,0 +1,162 @@
+} else {
+	checkDependenciesMatch( expectedDependencies, packageJsons );
+}
+
+/**
+ * @param {Object.<String, String>} expectedDependencies
+ * @param {Array.<Object>} packageJsons
+ * @param {Object.<String, String>} pathMappings
+ */
+function fixDependenciesVersions( expectedDependencies, packageJsons, pathMappings ) {
+	packageJsons
+		.filter( packageJson => packageJson.dependencies )
+		.forEach( packageJson => {
+			Object.entries( packageJson.dependencies )
+				.forEach( ( [ dependency, version ] ) => {
+					if ( version !== expectedDependencies[ dependency ] ) {
+						packageJson.dependencies[ dependency ] = expectedDependencies[ dependency ];
+					}
+				} );
+
+			fs.writeJsonSync( pathMappings[ packageJson.name ], packageJson, { spaces: 2 } );
+		} );
+
+	console.log( chalk.green( '✅  All dependencies fixed!' ) );
+}
+
+/**
+ * @param {Object.<String, String>} expectedDependencies
+ * @param {Array.<Object>} packageJsons
+ */
+function checkDependenciesMatch( expectedDependencies, packageJsons ) {
+	const errors = packageJsons
+		.filter( packageJson => packageJson.dependencies )
+		.flatMap( packageJson => Object.entries( packageJson.dependencies )
+			.map( ( [ dependency, version ] ) => {
+				if ( version !== expectedDependencies[ dependency ] ) {
+					return getWrongVersionErrorMsg( dependency, packageJson.name, version, expectedDependencies );
+				}
+			} )
+			.filter( Boolean )
+		);
+
+	if ( errors.length ) {
+		console.error( chalk.red( '❌  Errors found. Run this script with an argument: `--fix` to resolve the issues automatically:' ) );
+		console.error( chalk.red( errors.join( '\n' ) ) );
+		process.exit( 1 );
+	}
+
+	console.log( chalk.green( '✅  All dependencies are correct!' ) );
+}
+
+/**
+ * @param {String} dependency
+ * @param {String} name
+ * @param {String} version
+ * @param {Object.<String, String>} expectedDependencies
+ * @return {String}
+ */
+function getWrongVersionErrorMsg( dependency, name, version, expectedDependencies ) {
+	return `"${ dependency }" in "${ name }" in version "${ version }" should be set to "${ expectedDependencies[ dependency ] }".`;
+}
+
+/**
+ * @param {Array.<Object>} packageJsons
+ * @return {Object.<String, String>} expectedDependencies
+ */
+function getExpectedDepsVersions( packageJsons ) {
+	return packageJsons
+		.map( packageJson => packageJson.dependencies )
+		.filter( Boolean )
+		.reduce( ( expectedDependencies, dependencies ) => {
+			Object.entries( dependencies ).forEach( ( [ dependency, version ] ) => {
+				expectedDependencies[ dependency ] = getNewestVersion( version, expectedDependencies[ dependency ], dependency );
+			} );
+
+			return expectedDependencies;
+		}, {} );
+}
+
+/**
+ * @param {String|undefined} newVersion
+ * @param {String|undefined} currentMaxVersion
+ * @param {String} packageName
+ * @return {String}
+ */
+function getNewestVersion( newVersion = '0.0.0', currentMaxVersion = '0.0.0', packageName ) {
+	if ( newVersion.includes( '^' ) ) {
```

Comment:
This only covers one type of a range. Instead, we can use `semver` to check if the version is valid. The `semver.valid()` method returns `null` for any range.

```suggestion
	if ( !semver.valid( newVersion ) ) {
```